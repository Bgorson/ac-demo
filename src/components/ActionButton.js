import React from "react";
import PropTypes from 'prop-types';
import Button from "@material-ui/core/Button";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import ClickAwayListener from "@material-ui/core/ClickAwayListener";
import Grow from "@material-ui/core/Grow";
import Paper from "@material-ui/core/Paper";
import Popper from "@material-ui/core/Popper";
import MenuItem from "@material-ui/core/MenuItem";
import MenuList from "@material-ui/core/MenuList";
import { makeStyles } from "@material-ui/core/styles";

const options = ["Remove", "Email"];
const useStyles = makeStyles(() => ({
  mainButton: {
    padding: 15,
    fontSize: 12,
    height: "28px",
    backgroundColor: "white",
    textTransform: "none",
  },
  arrowButton: {
    fontSize: 12,
    minWidth: 10,
    padding: 0,
    backgroundColor: "white",
  },
  buttonDisplay: {
    visibility: "visible",
  },
  buttonHide: {
    visibility: "hidden",
  },
}));
export default function ActionButton({ isItemSelected }) {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const anchorRef = React.useRef(null);
  const [selectedIndex, setSelectedIndex] = React.useState(1);

  const handleClick = (e) => {
    e.stopPropagation();
  };

  const handleMenuItemClick = (event, index) => {
    event.stopPropagation();
    setSelectedIndex(index);
    setOpen(false);
  };

  const handleToggle = (e) => {
    e.stopPropagation();
    setOpen((prevOpen) => !prevOpen);
  };

  const handleClose = (event) => {
    event.stopPropagation();
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <ButtonGroup
        className={isItemSelected ? classes.buttonDisplay : classes.buttonHide}
        classvariant="contained"
        ref={anchorRef}
        aria-label="split button"
      >
        <Button className={classes.mainButton} onClick={handleClick}>
          {options[selectedIndex]}
        </Button>
        <Button
          className={classes.arrowButton}
          aria-controls={open ? "split-button-menu" : undefined}
          aria-expanded={open ? "true" : undefined}
          aria-label="select merge strategy"
          aria-haspopup="menu"
          onClick={(e) => handleToggle(e)}
        >
          <ArrowDropDownIcon />
        </Button>
      </ButtonGroup>
      <Popper
        style={{
          zIndex: 7000,
        }}
        open={open}
        anchorEl={anchorRef.current}
        role={undefined}
        transition
        disablePortal
      >
        {({ TransitionProps, placement }) => (
          <Grow
            {...TransitionProps}
            style={{
              transformOrigin:
                placement === "bottom" ? "center top" : "center bottom",
            }}
          >
            <Paper>
              <ClickAwayListener onClickAway={handleClose}>
                <MenuList id="split-button-menu">
                  {options.map((option, index) => (
                    <MenuItem
                      key={option}
                      disabled={index === 2}
                      selected={index === selectedIndex}
                      onClick={(event) => handleMenuItemClick(event, index)}
                    >
                      {option}
                    </MenuItem>
                  ))}
                </MenuList>
              </ClickAwayListener>
            </Paper>
          </Grow>
        )}
      </Popper>
    </div>
  );
}
ActionButton.propTypes = {
  isItemSelected: PropTypes.bool
};