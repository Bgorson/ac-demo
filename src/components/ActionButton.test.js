import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import ActionButton from './ActionButton';


it('the menu is only displayed when the row is selected', ()=>{
  const hidden = renderer.create(<ActionButton isItemSelected={false}  />).toJSON();
  expect(hidden).toMatchSnapshot();
  const shown = renderer.create(<ActionButton isItemSelected={true}  />).toJSON();
  expect(shown).toMatchSnapshot();
})
it('opens the menu when side button is clicked', async ()=>{
  render(<ActionButton isItemSelected={true} />)
  fireEvent.click(screen.getByTestId('menuArrow'))
  expect(screen.getByText('Remove')).toBeTruthy()

})