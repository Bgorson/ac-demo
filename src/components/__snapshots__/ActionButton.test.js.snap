// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the menu is only displayed when the row is selected 1`] = `
<div>
  <div
    aria-label="split button"
    className="MuiButtonGroup-root makeStyles-buttonHide-4"
    classvariant="contained"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-mainButton-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Email
      </span>
    </button>
    <button
      aria-haspopup="menu"
      aria-label="select merge strategy"
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-arrowButton-2"
      data-testid="menuArrow"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`the menu is only displayed when the row is selected 2`] = `
<div>
  <div
    aria-label="split button"
    className="MuiButtonGroup-root makeStyles-buttonDisplay-3"
    classvariant="contained"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-mainButton-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Email
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-haspopup="menu"
      aria-label="select merge strategy"
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-arrowButton-2"
      data-testid="menuArrow"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
